src/tdastro/sources/periodic_variable_star.py

```diff
@@ -0,0 +1,240 @@
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+from tdastro.astro_utils.black_body import black_body_luminosity_density_per_solid
+from tdastro.consts import PARSEC_TO_CM
+from tdastro.sources.periodic_source import PeriodicSource
+
+
+class PeriodicVariableStar(PeriodicSource, ABC):
+    """A model for a periodic variable star.
+
+    Attributes
+    ----------
+    period : `float`
+        The period of the source, in days.
+    epoch : `float`
+        The epoch of the zero phase, date.
+    distance : `float`
+        The distance to the source, in pc.
+    """
+
+    def __init__(self, period, epoch, **kwargs):
+        distance = kwargs.pop("distance", None)
```

Comment:
If I don't delete it I have a problem with duplicate parameter. Maybe I misuse the interface, please tell me how to implement it properly.

Reply:
That make sense. I think this is a bug how the interface works or at least how it is reporting the error. A superclass of `PeriodicVariableStar` (specifically `PhysicalModel`) already has an attribute `distance`. In this case it means the same thing, so you could either set it here or let it fall through to the `PhysicalModel` constructor. Either should give you the results you want. 

Because we draw the parameters for superclasses from kwargs, I added the duplicate check to make sure that we were not using the same name for multiple different variables.  For example if you used `distance` to indicate the distance between the stars in `EclipsingBinaryStar`, we'd want to prevent that to overriding distance to the source.